File: scrape_tmdb.py
import urllib.request
import re
import urllib.parse
import sys

movies = [
    ("Stranger Things 5", "Stranger Things"),
    ("Vingadores: Doomsday", "Avengers: Doomsday"),
    ("Avatar: Fogo e Cinzas", "Avatar: Fire and Ash"),
    ("Superman", "Superman 2025"),
    ("The Last of Us: Parte II", "The Last of Us"),
    ("Duna: Messias", "Dune Messiah"),
    ("Mandalorian & Grogu", "The Mandalorian & Grogu"),
    ("Missão Impossível 8", "Mission: Impossible - The Final Reckoning"),
    ("Casa do Dragão T3", "House of the Dragon"),
    ("Quarteto Fantástico", "The Fantastic Four: First Steps")
]

headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'}

for br_name, query in movies:
    q = urllib.parse.quote(query)
    url = f"https://www.themoviedb.org/search?query={q}"
    req = urllib.request.Request(url, headers=headers)
    try:
        html = urllib.request.urlopen(req).read().decode('utf-8')
        # We look for /t/p/w.../something.jpg
        # Example from tmdb search result: class="poster" src="/t/p/w94_and_h141_bestv2/xyz.jpg" or data-src=...
        
        # we just want the poster path: /abcde.jpg
        # Let's find all jpgs
        paths = re.findall(r'/t/p/[^/]+(/[^"]+\.jpg)', html)
        if paths:
            # Paths might be duplicates, just get the first valid looking one
            # The first search result poster is usually what we want
            print(f"{br_name}: https://image.tmdb.org/t/p/w500{paths[0]}")
        else:
            print(f"{br_name}: Not found")
            
    except Exception as e:
        print(f"Error for {br_name}: {e}")

File: scrape_tmdb.js
const movies = [
    { br_name: "Stranger Things 5", query: "Stranger Things" },
    { br_name: "Vingadores: Doomsday", query: "Avengers: Doomsday" },
    { br_name: "Avatar: Fogo e Cinzas", query: "Avatar: Fire and Ash" },
    { br_name: "Superman", query: "Superman" },
    { br_name: "The Last of Us: Parte II", query: "The Last of Us" },
    { br_name: "Duna: Messias", query: "Dune Messiah" },
    { br_name: "Mandalorian & Grogu", query: "The Mandalorian & Grogu" },
    { br_name: "Missão Impossível 8", query: "Mission: Impossible - The Final Reckoning" },
    { br_name: "Casa do Dragão T3", query: "House of the Dragon" },
    { br_name: "Quarteto Fantástico", query: "The Fantastic Four: First Steps" }
];

const headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
};

async function scrape() {
    for (const m of movies) {
        const url = `https://www.themoviedb.org/search?query=${encodeURIComponent(m.query)}`;
        try {
            const res = await fetch(url, { headers });
            const html = await res.text();

            // Look for the first poster image path.
            // In TMDB search results, posters are inside <img class="poster" ... src="/t/p/w.../xxxx.jpg"> or data-src
            const regex = /\/t\/p\/w[^/]+(\/[^"]+\.jpg)/g;
            const matches = [...html.matchAll(regex)];

            if (matches.length > 0) {
                // Return the first match's core path (group 1)
                console.log(`${m.br_name}: https://image.tmdb.org/t/p/w500${matches[0][1]}`);
            } else {
                console.log(`${m.br_name}: Not found`);
            }

            // Small delay to avoid triggering rate limit immediately
            await new Promise(r => setTimeout(r, 1000));
        } catch (e) {
            console.error(`Error for ${m.br_name}: ${e.message}`);
        }
    }
}

scrape();
